python/ray/dag/compiled_dag_node.py

```diff
@@ -943,8 +927,31 @@ def _preprocess(self) -> None:
                 task.arg_type_hints.append(upstream_task.dag_node.type_hint)
 
                 if upstream_task.dag_node.type_hint.requires_nccl():
-                    # Add all readers to the NCCL actors of P2P.
-                    nccl_actors_p2p.add(downstream_actor_handle)
+                    custom_nccl_group: Optional[
+                        GPUCommunicator
+                    ] = upstream_task.dag_node.type_hint.get_custom_nccl_group()
+                    if custom_nccl_group is None and self._custom_nccl_group_p2p:
+                        custom_nccl_group = self._custom_nccl_group_p2p
+                    if custom_nccl_group:
+                        # A custom NCCL group is used. Check it contains both
+                        # upstream and downstream actors.
+                        nccl_group_actors = custom_nccl_group.get_actor_handles()
+                        assert downstream_actor_handle
+                        upstream_actor_handle: Optional[
```

Comment:
Nit: remove type hint for `upstream_actor_handle`.

Reply:
Done.